app/src/main/java/org/dhis2/Bindings/EventExtensions.kt

```diff
@@ -0,0 +1,19 @@
+package org.dhis2.Bindings
+
+import org.hisp.dhis.android.core.event.Event
+import org.hisp.dhis.android.core.event.EventStatus
+import java.util.Date
+
+fun Event.primaryDate() : Date {
+
+    return when(status()){
+        EventStatus.ACTIVE -> eventDate()!!
+        EventStatus.COMPLETED -> eventDate()!!
+        EventStatus.SCHEDULE -> dueDate()!!
+        EventStatus.SKIPPED -> dueDate()!!
+        EventStatus.VISITED -> eventDate()!!
+        EventStatus.OVERDUE -> dueDate()!!
+        null -> Date()
```

Comment:
What if eventDate() or dueDate() are null?

Reply:
Can't be both null. All Active, completed and visited events have an eventDate. And any schedule, skipped and overdue have a dueDate.
It is a problem in the sdk with wrong @Nullable annotations